src/test/java/de/tum/in/www1/artemis/TextExerciseIntegrationTest.java

```diff
@@ -129,6 +129,37 @@ public void createTextExercise() throws Exception {
         assertThat(newTextExercise.getCourseViaExerciseGroupOrCourseMember().getId()).as("exerciseGroupId was set correctly").isEqualTo(course.getId());
     }
 
+    @Test
+    @WithMockUser(value = "instructor1", roles = "INSTRUCTOR")
+    public void createTextExercise_setExerciseTitleNull_badRequest() throws Exception {
+        final Course course = database.addCourseWithOneReleasedTextExercise();
+        TextExercise textExercise = textExerciseRepository.findByCourseId(course.getId()).get(0);
```

Comment:
Here you want to test the creation of a new TextExercise, so I would just create a new one with the constructor, instead of fetching one from the repository. Also, this test case (from what I can see) is supposed to cover the case where the title is set to null, but since you're trying to save an exercise that you're getting from the repository, that exercise would already have an id. If you look at the corresponding endpoint in the TextExerciseResource class, the first if statement will catch the request you're sending (since the id is set, and it shouldn't be for this request) and return before it even reaches the second if statement that checks the title (the one you would like to test).

Reply:
ahhh, I see... I did not notice that at first. Thank you, I'll change like that